library/dnet.c

```diff
@@ -761,14 +761,6 @@ static int dnet_cmd_bulk_read(struct dnet_backend_io *backend, struct dnet_net_s
 		cmd->flags &= ~DNET_FLAGS_NEED_ACK;
 	}
 
-	/*
-	 * we have to drop io lock, otherwise it will be grabbed again in dnet_process_cmd_raw() being recursively called
-	 * Lock will be taken again after loop has been finished
-	 */
-	if (!(cmd->flags & DNET_FLAGS_NOLOCK)) {
-		dnet_opunlock(st->n, &cmd->id);
```

Comment:
Why did you remove it? `cmd->id` coincides with one of ids from ios, so it should be unlocked before calling `dnet_process_cmd_raw` and should be locked after.